src/App.tsx

```diff
@@ -1,61 +1,110 @@
+import { useState } from 'react';
+
 import './App.scss';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+
+export const App: React.FC = () => {
+  const [curentTodos, setCurentTodos] = useState(todosFromServer);
+
+  const [newTitle, setNewTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [choosenUser, setChoosenUser] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const maxId = (arr: Todo[]): number => {
+    const max = Math.max(...arr.map(object => object.id));
+
+    return max + 1;
+  };
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+  const reset = () => {
+    setNewTitle('');
+    setChoosenUser(0);
+    setHasTitleError(false);
+    setHasUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setHasTitleError(!newTitle);
+    setHasUserError(choosenUser === 0);
+
+    if (!newTitle || choosenUser === 0) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: maxId(curentTodos),
+      title: newTitle,
+      completed: false,
+      userId: choosenUser,
+    };
+
+    setCurentTodos([...curentTodos, newTodo]);
+
+    reset();
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title-input">Title: </label>
+          <input
+            id="title-input"
+            type="text"
+            data-cy="titleInput"
+            value={newTitle}
+            placeholder="Enter a title"
+            onChange={(event) => {
+              setNewTitle(event.target.value);
+              setHasTitleError(false);
+            }}
```

Comment:
If you want, you can create a separate function for onChange, and pass it here for better readability:
```ts
const onInputChange = (event) => {
  setNewTitle(event.target.value);
  setHasTitleError(false);
}
```

Reply:
and pass it like this:
```ts
onChange={onInputChange}
```